From 1487b5b44c97e6ada571d6e229a8664934e4de11 Mon Sep 17 00:00:00 2001
From: Avi Niddam <aniddam@habana.ai>
Date: Thu, 05 Jun 2025 13:53:29 +0300
Subject: [PATCH] [SW-228856] fit: use only temporary fct

Context:
We encountered several code duplications related to
fit_foreach_property.
So, we want to create a new fit services infrastructure
to solve this issue in a clean way.
The main Lego pieces are the newly created files:
fit_services.c/h

Current commit:
In this current commit, we remove the function
fit_foreach_property from ft.c/h and temporarily use
the function _fit_foreach_property.

General remark:
This infrastructure is intended to support ANY of the
fit related functions that we find today in fit_utils.c
or fit.c
Therefore, from now, please, anyone doing a change related
to the fit files should carefully double-check:
Any unified code can be (and should be) located in
fit_services.c/h
For this, please, follow the example of how it has been
done with fit_foreach_property.

Change-Id: I5e88e8e5071c811937f97ff939a672efe6a41a9e
Signed-off-by: Avi Niddam <aniddam@habana.ai>
---

diff --git a/lib/preboot/fit.c b/lib/preboot/fit.c
index cd7da3a..8468a7c 100644
--- a/lib/preboot/fit.c
+++ b/lib/preboot/fit.c
@@ -36,6 +36,7 @@
 #include <lib_common.h>
 #include <oob_if.h>
 #include <emb_cmn_log.h>
+#include "fit_services.h"
 
 #ifdef __ZEPHYR__
 LOG_MODULE_REGISTER(fit_emb_common);
@@ -216,34 +217,6 @@
 	return dflt_conf_node;
 }
 
-int fit_foreach_property(const void *fit, const char *node_path,
-			fit_property_match_cb match_cb, void *data)
-{
-	int ret, count = 0;
-	int noffset, property;
-	const struct fdt_property *prop;
-
-	noffset = fdt_path_offset(fit, node_path);
-	if (noffset < 0) {
-		EMB_LOG_RAW_ERR("Cannot find entry ");
-		EMB_LOG_RAW(node_path);
-		EMB_LOG_RAW("\n\r");
-		return -1;
-	}
-
-	fdt_for_each_property_offset(property, fit, noffset) {
-		prop = fdt_get_property_by_offset(fit, property, NULL);
-		if (!prop)
-			continue;
-
-		ret = match_cb(fit, prop, data);
-		if (!ret)
-			count++;
-	}
-
-	return count;
-}
-
 int match_prop_version(const void *fit, const struct fdt_property *prop, void *data)
 {
 	const char *cmpstr = (const char *)data;
@@ -266,7 +239,7 @@
 {
 	int ret;
 
-	ret = fit_foreach_property(fit, "/", match_prop_version, "version");
+	ret = _fit_foreach_property(fit, "/", match_prop_version, "version");
 	if (ret <= 0)
 		EMB_LOG_WRN("[FIT] No version properties were found.");
 }
diff --git a/lib/preboot/fit.h b/lib/preboot/fit.h
index b07d3be..7933b62 100644
--- a/lib/preboot/fit.h
+++ b/lib/preboot/fit.h
@@ -70,9 +70,6 @@
 		       size_t sec_desc_size, const void *pub_key_fit,
 		       const u8 *sig, size_t sig_len);
 
-typedef int (*fit_property_match_cb)(const void *fit, const struct fdt_property *prop, void *data);
-int fit_foreach_property(const void *fit, const char *node_path, fit_property_match_cb match_cb,
-						void *data);
 void fit_print_versions(void *fit);
 int fit_check_format(const void *fit);
 bool is_fit_verify_supported(void);
